src/lib/logical_query_plan/lqp_utils.cpp

```diff
@@ -231,45 +231,75 @@ std::set<std::string> lqp_find_modified_tables(const std::shared_ptr<AbstractLQP
   return modified_tables;
 }
 
-std::shared_ptr<AbstractExpression> lqp_subplan_to_boolean_expression(
-    const std::shared_ptr<AbstractLQPNode>& begin, const std::optional<const std::shared_ptr<AbstractLQPNode>>& end) {
+namespace {
+/**
+ * Function creates a boolean expression from an lqp whilst preserving the same order of how expressions are evaluated.
```

Comment:
I approve the change of this PR, but omg, this comment block is soooo long, if there is any important info in it noone will ever notice it. Can you maybe shorten it a bit?

Reply:
I shortened it.